### PR TITLE
Correctly pass Pod::VERSION during pod lib create

### LINKED ISSUE #5840: pod lib create fails
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod lib create <MyLibraryName>` in an empty directory

## What did you expect to happen?

Run the pod template installer

### Error

```
TypeError - no implicit conversion of Class into String
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:78:in `system'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:78:in `block (2 levels) in configure_template'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:76:in `chdir'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:76:in `block in configure_template'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/user_interface.rb:64:in `section'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:75:in `configure_template'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command/lib/create.rb:42:in `run'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/lib/cocoapods/command.rb:50:in `run'
/Users/narciero/.rvm/gems/ruby-2.2.1@global/gems/cocoapods-1.1.0.rc.1/bin/pod:55:in `<top (required)>'
/Users/narciero/.rvm/gems/ruby-2.2.1/bin/pod:23:in `load'
/Users/narciero/.rvm/gems/ruby-2.2.1/bin/pod:23:in `<main>'
/Users/narciero/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `eval'
/Users/narciero/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `<main>'
```

A project folder does get created though.

### Stack

```
   CocoaPods : 1.1.0.rc.1
        Ruby : ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]
    RubyGems : 2.4.6
        Host : Mac OS X 10.11.6 (15G1004)
       Xcode : 8.0 (8A218a)
         Git : git version 2.8.4 (Apple Git-73)
Ruby lib dir : /Users/narciero/.rvm/rubies/ruby-2.2.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ f5d850fac58a70d7ac421b1c02ea2733118d3a59
```

### Installation Source

```
Executable Path: /Users/narciero/.rvm/gems/ruby-2.2.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### PR DESCRIPTION
**Pull request by DanToml**
closes #5840

### CODE REVIEW

**Comment by segiddins** [comment]
Is is possible to add a test for this?

**Comment by orta** [comment]
Nice one 👍

**Comment by orta** [comment]
Somewhere we're still expecting a `Pod::Version`
```
- expected exactly once, not yet invoked: #<AnyInstance:Pod::Command::Lib::Create>.system({'COCOAPODS_VERSION' => Pod::Version}, './configure', 'TestPod', 'foo')
```

File: lib/cocoapods/command/lib/create.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def configure_template
           UI.section("Configuring #{@name} template.") do
             Dir.chdir(@name) do
               if File.exist?('configure')
-                system({ 'COCOAPODS_VERSION' => Pod::Version }, './configure', @name, *@additional_args)
+                system({ 'COCOAPODS_VERSION' => Pod::VERSION }, './configure', @name, *@additional_args)
               else
```
Comment (orta) [inline lib/cocoapods/command/lib/create.rb]: There is a spec for this - https://github.com/CocoaPods/CocoaPods/blob/master/spec/functional/command/lib/create_spec.rb#L37

Comment (orta) [inline lib/cocoapods/command/lib/create.rb]: that's what's failing

Comment (DanToml) [inline lib/cocoapods/command/lib/create.rb]: Oh didn't see that 👍  ta

**Comment by orta** [comment]
I've removed the new spec file, as it was a dupe - and fixed the existing test 👍

**Comment by DanToml** [comment]
@orta thanks!

**Pull request merged**